pkg/transform/oauth/basicauth.go

```diff
@@ -12,18 +12,30 @@ import (
 // IdentityProviderBasicAuth is a basic auth specific identity provider
 type IdentityProviderBasicAuth struct {
 	identityProviderCommon `yaml:",inline"`
-	BasicAuth              struct {
-		URL string `yaml:"url"`
-		CA  struct {
-			Name string `yaml:"name"`
-		} `yaml:"ca"`
-		TLSClientCert struct {
-			Name string `yaml:"name"`
-		} `yaml:"tlsClientCert"`
-		TLSClientKey struct {
-			Name string `yaml:"name"`
-		} `yaml:"tlsClientKey"`
-	} `yaml:"basicAuth"`
+	BasicAuth              BasicAuth `yaml:"basicAuth"`
+}
+
+// BasicAuth provider specific data
+type BasicAuth struct {
+	URL           string                 `yaml:"url"`
+	CA            BasicAuthCA            `yaml:"ca"`
+	TLSClientCert BasicAuthTLSClientCert `yaml:"tlsClientCert"`
+	TLSClientKey  BasicAuthTLSClientKey  `yaml:"tlsClientKey"`
+}
+
+// BasicAuthCA holds basic auth CA Name
+type BasicAuthCA struct {
+	Name string `yaml:"name"`
+}
```

Comment:
I would actually add a `types.go` in the `oauth` directory that contains the following:
```golang
type CA struct {
    Name string `yaml:"name"`
}
```

Because all of the CA's I see in the structs below look identical. I don't see the need for having a `BasicAuthCA`, `LdapCA`, `GitHubCA`, etc. They should just be `CA`. Or you could use the fullname for the struct `CertificateAuthority` or `OauthCA` but that's going to cause the stutter error `oauth.OauthCA`.